### ISSUE TITLE
Default client ID set by examples is truncated to non-unique value

### ISSUE THREAD

**Issue by tdicola** [adafruit/Adafruit_MQTT_Library#4]
We noticed when running two instances of an example that they're colliding over the same client ID.  Right now the logic is to concat AIO key + compile date & time to get a unique value for the user, however if two examples compiled on the same day run then one of them will knock the other off the broker.  Likely the client ID is being truncated on the server side and only the first xx chars are being used (before the date).  Let's consider a quick fix to swap the order so date and time come first and then AIO key, and a longer term fix to make this more robust on the server side.

**Comment by Sushinn** [comment]
what I do on another MQTT library that I use, is use the Mac address as the client ID. However I have not tried to use more than one instance so far. (waiting for my ESP-01s and 07s to arrive)

**Comment by toddtreece** [comment]
I like the MAC address idea. Seems like it would work well for Arduino based projects since you most likely won't be running multiple MQTT clients on one device.

The code that is limiting the length of the client ID can be found here:
https://github.com/adafruit/Adafruit_MQTT_Library/blob/master/Adafruit_MQTT.cpp#L308

It's actually part of the MQTT v3.1 spec to limit the client ID to 23 chars, so maybe we should just add support for v3.1.1 so we remove that restriction and get the auto-generated client ID feature. The other option is just dropping that length restriction for now. The Adafruit IO MQTT broker currently supports really long client IDs, but it could cause some connection weirdness on strict MQTT v3.1 brokers.

**Comment by toddtreece** [comment]
this should be fixed by #10 & the release of Adafruit_MQTT_Library v0.12.0. you can now leave client id out of the list of arguments, and the server will generate a unique ID for your client.

```
Adafruit_MQTT_FONA mqtt(&fona, MQTT_SERVER, AIO_SERVERPORT, MQTT_USERNAME, MQTT_PASSWORD);
```
